pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -362,11 +380,25 @@ type KubernetesSDConfig struct {
 	// If left empty, Prometheus is assumed to run inside
 	// of the cluster. It will discover API servers automatically and use the pod's
 	// CA certificate and bearer token file at /var/run/secrets/kubernetes.io/serviceaccount/.
+	// +kubebuilder:validation:Minlength=1
 	// +optional
 	APIServer *string `json:"apiServer,omitempty"`
 	// Role of the Kubernetes entities that should be discovered.
 	// +required
-	Role Role `json:"role"`
+	Role KubernetesRole `json:"role"`
+	// Optional namespace discovery. If omitted, Prometheus discovers targets across all namespaces.
+	// +optional
+	Namespaces *NamespaceDiscovery `json:"namespaces,omitempty"`
+	// Optional metadata to attach to discovered targets.
+	// It requires Prometheus >= v2.35.0 for `pod` role and
+	// Prometheus >= v2.37.0 for `endpoints` and `endpointslice` roles.
+	// +optional
+	AttachMetadata *AttachMetadata `json:"attachMetadata,omitempty"`
+	// Selector to select objects.
```

Comment:
Can you document which versions are required?

Reply:
Unsure what you mean here.

Reply:
Not all versions of Prometheus support all selector combinations.

Reply:
I mean Prometheus >= 2.17.0